code401challenges/src/test/java/code401challenges/sort/InsertionSortTest.java

```diff
@@ -0,0 +1,43 @@
+package code401challenges.sort;
+
+import org.junit.Test;
+        import static org.junit.Assert.*;
+
+
+public class InsertionSortTest {
+    public class InsertionSorter {
+        public void sort(int[] values) {
+            for (int i = 1; i < values.length; ++i) {
+                insert(values, i-1, values[i]);
+            }
+        }
+        private void insert(int[] array, int rightIndex, int value) {
+            while (0 <= rightIndex && value < array[rightIndex]) {
+                array[rightIndex+1] = array[rightIndex];
+                --rightIndex;
+            }
+            array[rightIndex+1] = value;
+        }
+    }
```

Comment:
You also wrote sort code in this testing file, and that is the code that you are actually testing. The InsertionSort code should be in a different file.